### ISSUE TITLE
Overwrite cy.request/cy.visit - calling original function more than once?

### ISSUE THREAD

**Issue by MarcBridner** [cypress-io/cypress#9174]
I'm trying to overwrite the "request" (and visit) commands to do authentication verification before each request/visit (for reasons ... )

My first attempt was this:
```js
Cypress.Commands.overwrite('request', (originalFn, url, options) => {
  originalFn('/sso/login')
    .then(response => {
      // do stuff here depending on data in response
      cy.log(response);
    })
    .then(() => originalFn(url, options));
});
```

The first request to /sso/login doesn't seem to complete, when looking in debug console it looks like this:

```
Command:  request
Request:  []
Yielded:  {}
```
Which is not the correct response, the payload is JSON and always has something in it. Additionally, the last/original call doesn't seem to complete either, cypress just hangs there.

I've tried all variations of this I can think of, including cy.wrap, cy.window and various other hacks. Most of them give me the promise in promise warning. changing the first line from originalFn("/sso/login") to cy.request('/sso/login") causes cypress to hang, my guess it goes into an infinite recursion.

Cypress 3.5.0 if it matters, but I don't think much has changed here

Any tips? What am I doing wrong?

**Comment by jennifer-shehane** [comment]
In the example given, you are calling the originalFn twice, so isn't it expected that it would call it twice? 

In your case, you likely just want a custom command that contains a series of Cypress commands to run, as it seems you are not actually changing the way the `cy.request()` command works (which would be the primary reason to overwrite a command). https://on.cypress.io/custom-commands#Parent-Commands 

I think trying to overwrite the command is causing these unintended consequences for you. Can you try something like below with `Cypress.Command.add()`

```js
Cypress.Commands.add('loginRequest', (label) => {
  cy.request('/sso/login')
  .then(response => {
    // do stuff here depending on data in response
    cy.log(response);
  })
})

// later in test
cy.loginRequest()
```

**Comment by MarcBridner** [comment]
Yes, i expected the underlying function to be called twice and complete successfully and correctly. It doesn't.

I wanted to do authentication on request/visit transparently by overrwriting them, as opposed to adding requestAuthed and visitAuthed commands. I did end up adding commands since overwrites didn't behave as expected when the original function was called twice, but I figured that I'd create an issue and it may be fixed some day.
